### PR TITLE
UICHKIN-398 - Hide 'new fee/fine' menu item for dcb user

### PR DESCRIPTION
**Pull request by Terala-Priyanka**
## Purpose
[DCB] Check-in: Hide Fee/Fine action menu options (Lending library)

## Refs
[UICHKIN-398](https://issues.folio.org/browse/UICHKIN-398)

## Pre-Merge Checklist
Before merging this PR, please go through the following list and take appropriate actions.
- [x] I've added appropriate record to the CHANGELOG.md
- Does this PR meet or exceed the expected quality standards?
  - [x] Code coverage on new code is 80% or greater
  - [x] Duplications on new code is 3% or less
  - [x] There are no major code smells or security issues
- Does this introduce breaking changes?
  - [ ] If any API-related changes - okapi interfaces and permissions are reviewed/changed correspondingly
  - [x] There are no breaking changes in this PR.

If there are breaking changes, please **STOP** and consider the following:

- What other modules will these changes impact?
- Do JIRAs exist to update the impacted modules?
  - [ ] If not, please create them
  - [ ] Do they contain the appropriate level of detail?  Which endpoints/schemas changed, etc.
  - [ ] Do they have all they appropriate links to blocked/related issues?
- Are the JIRAs under active development?
  - [ ] If not, contact the project's PO and make sure they're aware of the urgency.
- Do PRs exist for these changes?
  - [ ] If so, have they been approved?

Ideally all of the PRs involved in breaking changes would be merged in the same day to avoid breaking the folio-testing environment.  Communication is paramount if that is to be achieved, especially as the number of intermodule and inter-team dependencies increase.

While it's helpful for reviewers to help identify potential problems, ensuring that it's safe to merge is ultimately the responsibility of the PR assignee.

### CODE REVIEW

File: src/consts.js [hunk 1]
```diff
@@ -56,3 +56,7 @@ export const cancelFeeClaimReturned = {
 };
 
 export const MAX_RECORDS = '1000';
+
+export const DCB_USER = {
```
Comment (Dmitriy-Litvinenko) [inline src/consts.js:60]: Could you please clarify this user will be created to BE and always have the same ID?

Comment (Terala-Priyanka) [reply src/consts.js:60]: Basically, these are the virtual patrons that are created by BE scripts only.
There can be many users with lastName "DcbSystem" and type "dcb".
Here we need check for user type as it is not obtained in the checkin response.

**Review by Dmytro-Melnyshyn — APPROVED** [review]

File: src/util.test.js [hunk 2]
```diff
@@ -318,3 +320,18 @@ describe('shouldConfirmStatusModalBeShown', () => {
...
 });
+
+describe('isDcbUser', () => {
+  it('should return true when user has lastName as "DcbSystem"', () => {
+    const user = {
+      lastName: DCB_USER.lastName
+    };
+    expect(isDcbUser(user)).toBeTruthy();
+  });
+  it('should return false when user does not have lastName as "DcbSystem"', () => {
```
Comment (Dmitriy-Litvinenko) [inline src/util.test.js:331]: Could you please add empty line before `it`?

File: src/util.test.js [hunk 3]
```diff
@@ -318,3 +320,18 @@ describe('shouldConfirmStatusModalBeShown', () => {
     });
   });
 });
+
+describe('isDcbUser', () => {
+  it('should return true when user has lastName as "DcbSystem"', () => {
+    const user = {
+      lastName: DCB_USER.lastName
+    };
+    expect(isDcbUser(user)).toBeTruthy();
```
Comment (Dmitriy-Litvinenko) [inline src/util.test.js:329]: Could you please add empty line before `expect`?

File: src/util.test.js [hunk 4]
```diff
@@ -318,3 +320,18 @@ describe('shouldConfirmStatusModalBeShown', () => {
...
+  });
+  it('should return false when user does not have lastName as "DcbSystem"', () => {
+    const user = {
+      lastName: 'test'
+    };
+    expect(isDcbUser(user)).toBeFalsy();
```
Comment (Dmitriy-Litvinenko) [inline src/util.test.js:335]: Could you please add empty line before `expect`?

File: src/util.test.js [hunk 5]
```diff
@@ -318,3 +320,18 @@ describe('shouldConfirmStatusModalBeShown', () => {
...
+describe('isDcbUser', () => {
+  it('should return true when user has lastName as "DcbSystem"', () => {
+    const user = {
+      lastName: DCB_USER.lastName
+    };
+    expect(isDcbUser(user)).toBeTruthy();
+  });
+  it('should return false when user does not have lastName as "DcbSystem"', () => {
+    const user = {
+      lastName: 'test'
```
Comment (Dmitriy-Litvinenko) [inline src/util.test.js:333]: Could you please add comma?

File: src/util.test.js [hunk 6]
```diff
@@ -318,3 +320,18 @@ describe('shouldConfirmStatusModalBeShown', () => {
     });
   });
 });
+
+describe('isDcbUser', () => {
+  it('should return true when user has lastName as "DcbSystem"', () => {
+    const user = {
+      lastName: DCB_USER.lastName
```
Comment (Dmitriy-Litvinenko) [inline src/util.test.js:327]: Could you please add comma?

File: src/util.js [hunk 7]
```diff
@@ -4,7 +4,7 @@ import {
   includes,
 } from 'lodash';
 
-import { statuses } from './consts';
+import { DCB_USER, statuses } from './consts';
```
Comment (Dmitriy-Litvinenko) [inline src/util.js:7]: Could you please split in several line?

File: src/util.test.js (hunk 2 above)
Comment (Terala-Priyanka) [reply src/util.test.js:333]: done

File: src/util.test.js (hunk 3 above)
Comment (Terala-Priyanka) [reply src/util.test.js:330]: done

File: src/util.test.js (hunk 4 above)
Comment (Terala-Priyanka) [reply src/util.test.js:338]: done

File: src/util.test.js (hunk 5 above)
Comment (Terala-Priyanka) [reply src/util.test.js]: done

File: src/util.test.js (hunk 6 above)
Comment (Terala-Priyanka) [reply src/util.test.js]: done

File: src/util.js (hunk 7 above)
Comment (Terala-Priyanka) [reply src/util.js]: done

**Comment by Terala-Priyanka** [comment]
@Dmitriy-Litvinenko 
Fixed all your comments. Also please note that I have added changes to couple more files - FeeFineDetailsButton.js and FeeFineDetailsButton.test.js
Please re-review

File: src/util.js [hunk 8]
```diff
@@ -4,7 +4,10 @@ import {
   includes,
 } from 'lodash';
 
-import { statuses } from './consts';
+import {
+  DCB_USER,
+  statuses
```
Comment (Dmitriy-Litvinenko) [inline src/util.js:9]: Could you please add comma?

File: src/components/FeeFineDetailsButton/FeeFineDetailsButton.js [hunk 9]
```diff
@@ -15,6 +15,7 @@ const FEE_FINES_STATUSES = {
 
 class FeeFineDetailsButton extends React.Component {
   static propTypes = {
+    isVirtualUser: PropTypes.bool,
```
Comment (Dmitriy-Litvinenko) [inline src/components/FeeFineDetailsButton/FeeFineDetailsButton.js:18]: Could you please clarify should we add `isRequired `or add default value?

Comment (Terala-Priyanka) [reply src/components/FeeFineDetailsButton/FeeFineDetailsButton.js:18]: added default prop value

File: src/util.js (hunk 8 above)
Comment (Terala-Priyanka) [reply src/util.js]: done

**Review by Dmytro-Melnyshyn — APPROVED** [review]

**Review by Dmitriy-Litvinenko — APPROVED** [review]

**Pull request merged**